test/intl402/DateTimeFormat/canonicalize-timezone.js

```diff
@@ -3,16 +3,94 @@
+          3. If _identifierCountryCode_ is _zoneCountryCode_, then
+            a. Set _primary_ to _zone_.
+          4. Else,
+            a. Let _countryCodeLineCount_ be the number of lines in file <code>zone.tab</code> of the IANA Time Zone Database where the “country-code” column is _identifierCountryCode_.
+            b. If _countryCodeLineCount_ is 1, then
+              i. Let _countryCodeLine_ be the line in file <code>zone.tab</code> of the IANA Time Zone Database where the “country-code” column is _identifierCountryCode_.
+              ii. Set _primary_ to the contents of the “TZ” column of _countryCodeLine_.
+            c. Else,
+              i. Let _backzone_ be *undefined*.
+              ii. Let _backzoneLinkLines_ be the List of lines in the file <code>backzone</code> of the IANA Time Zone Database that start with either *"Link "* or *"#PACKRATLIST zone.tab Link "*.
+              iii. ...
+              iv. Assert: _backzone_ is not *undefined*.
+              v. Set _primary_ to _backzone_.
+      c. If _primary_ is one of *"Etc/UTC"*, *"Etc/GMT"*, or *"GMT"*, set _primary_ to *"UTC"*.
+      d. ...
+      e. Let _record_ be the Time Zone Identifier Record { [[Identifier]]: _identifier_, [[PrimaryIdentifier]]: _primary_ }.
+      f. Append _record_ to _result_.
+    6. ...
+    7. Return _result_.
+
+    GetAvailableNamedTimeZoneIdentifier ( _timeZoneIdentifier_ )
+
+    1. For each element _record_ of AvailableNamedTimeZoneIdentifiers(), do
+      a. If _record_.[[Identifier]] is an ASCII-case-insensitive match for _timeZoneIdentifier_, return record.
+    2. Return ~empty~.
+
+    CreateDateTimeFormat ( _newTarget_, _locales_, _options_, _required_, _defaults_ )
+
+    29. If IsTimeZoneOffsetString(_timeZone_) is *true*, then
+      ...
+    30. Else,
+      a. Let _timeZoneIdentifierRecord_ be GetAvailableNamedTimeZoneIdentifier(_timeZone_).
+      b. If _timeZoneIdentifierRecord_ is ~empty~, throw a RangeError exception.
+      c. Set _timeZone_ to _timeZoneIdentifierRecord_.[[PrimaryIdentifier]].
 ---*/
 
 const timeZones = [
+  // Europe/Prague is not a Link name.
   ["Europe/Prague", "Europe/Prague"],
+
+  // `backward` identifies "Europe/Bratislava" as a Link name targeting "Europe/Prague":
+  //   Link	Europe/Prague		Europe/Bratislava
+  // Europe/Bratislava's country code "SK" has only one time zone in `zone.tab`.
   ["Europe/Bratislava", "Europe/Bratislava"],
+
+  // `backward` identifies "Australia/Canberra" as a Link targeting "Australia/Sydney":
+  //   Link	Australia/Sydney	Australia/Canberra
+  // Both share the country code "AU".
+  ["Australia/Canberra", "Australia/Sydney"],
+
+  // `backward` identifies "Atlantic/Jan_Mayen" as a Link name targeting "Europe/Berlin".
+  //   Link	Europe/Berlin		Atlantic/Jan_Mayen
+  // Atlantic/Jan_Mayen's country code "SJ" which has only one time zone in `zone.tab`.
```

Comment:
Whoops, my last suggestion had some formatting mistakes.
```suggestion
  // `backward` identifies "Atlantic/Jan_Mayen" as a Link name targeting "Europe/Berlin":
  //   Link	Europe/Berlin		Atlantic/Jan_Mayen
  // Atlantic/Jan_Mayen's country code "SJ" has only one time zone in `zone.tab`.
```

Reply:
Applied